packages/components/tool-bar/src/tool-bar.ts

```diff
@@ -0,0 +1,229 @@
+}
+
+export interface ToolBarItemGroup extends ToolBarItemBase {
+  type: 'group';
+  buttons: ToolBarItemButton[];
+  label?: string | null;
+  selects?: 'single' | 'multiple';
+}
+
+export type ToolBarItem = ToolBarItemButton | ToolBarItemDivider | ToolBarItemGroup;
+
+Icon.register(faEllipsisVertical);
+
+/**
+ * A responsive container that automatically hides items in an overflow menu when space is limited.
+ *
+ * @csspart wrapper - The wrapper element that contains the tool bar items.
+ *
+ * @slot - The tool bar items.
+ */
+export class ToolBar extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-icon': Icon,
+      'sl-menu-button': MenuButton,
+      'sl-menu-item': MenuItem,
+      'sl-menu-item-group': MenuItemGroup
+    };
+  }
+
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  // Observe changes to the size of the element.
+  #observer = new ResizeObserver(() => this.#onResize());
+
+  /** The horizontal alignment within the tool-bar. */
+  @property({ reflect: true }) align?: 'start' | 'end';
+
+  /** If true, the tool-bar is disabled and cannot be interacted with. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** @internal True when the tool-bar is empty. */
+  @property({ type: Boolean, reflect: true }) empty?: boolean;
+
+  /** @internal The tool bar items. */
+  @state() items: ToolBarItem[] = [];
+
+  /** @internal The tool bar items that should be shown in the overflow menu. */
+  @state() menuItems: ToolBarItem[] = [];
+
+  /**
+   * If true, will cause the tool bar to not have a border; useful when embedding
+   * the tool-bar inside another component.
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'no-border' }) noBorder?: boolean;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.setAttribute('role', 'toolbar');
+
+    this.#observer.observe(this);
+  }
+
+  override disconnectedCallback(): void {
+    this.#observer.disconnect();
+
+    super.disconnectedCallback();
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('disabled')) {
+      const children = this.renderRoot.querySelector('slot')?.assignedElements({ flatten: true }) ?? [];
+
+      if (this.disabled) {
+        children.forEach(el => el.setAttribute('disabled', ''));
+      } else {
+        children.forEach(el => el.removeAttribute('disabled'));
+      }
+    }
+
+    if (changes.has('items')) {
+      this.menuItems = this.items.filter(item => !item.visible);
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <div part="wrapper">
+        <slot @slotchange=${this.#onSlotChange}></slot>
+      </div>
+
+      ${this.menuItems.length
+        ? html`
+            <sl-menu-button>
+              <sl-icon name="far-ellipsis-vertical" slot="button"></sl-icon>
```

Comment:
`aria-label` is missing here? :)

Reply:
The aria-label should probably on the actual `<sl-button>`, but afaics, there is no way to set that atm. Something to fix.

Reply:
I've added a `FIXME` comment so we remember to do this